ui/src/app/pages/login/sign-in.tsx

```diff
@@ -403,6 +404,7 @@ export class SignInImpl extends React.Component<SignInProps, SignInState> {
   }
 
   private onSubmit = async () => {
+    const { enableCaptcha } = serverConfigStore.get().config;
```

Comment:
this file is the only UI consumer of the flag.  Use the API FF in `serverConfigStore` instead of the UI env-flag in `environment`